### ISSUE TITLE
Cannot extend classes defined in modules

### ISSUE THREAD

**Issue by flibX0r** [jac3km4/redscript#37]
Presumably due to the name mangling that makes modules work.
Fixable with a pass over all the classes in the pool and update their base class name?

**Example:**
If I define a module:
```swift
module BaseModule

public class BaseClass {
  public func Foo() -> Void { Log("Base!"); }
}
```

and then in another script attempt to extend it:
```swift
import BaseModule.BaseClass

public class DerivedClass extends BaseClass {
  public func Foo() -> Void { Log("Derived!");  }
}
```

redscript is happy to compile it but Cyberpunk throws an error:
![Cyberpunk2077_Error](https://user-images.githubusercontent.com/6767042/115970979-86bb1c00-a578-11eb-8c1f-1044b0f4d7d1.png)

**Comment by jac3km4** [comment]
I think this isn't related to modules, but the order in which classes are stored in the cache by the compiler. It seems that the game requires the base class to always appear before the subclass. This would require the compiler to sort classes before writing the cache file.
There's a workaround for this though - you'd need to reorder the class definitions/files so that the base class gets compiled first.

**Comment by jac3km4** [comment]
fix for this has been merged in 364d3d45ecac3d1f65551254febbbbf7bf9a2191
